subworkflows/local/prepare_genome.nf

```diff
@@ -118,6 +122,33 @@ workflow PREPARE_GENOME {
     //
     ch_chrom_sizes = GET_CHROM_SIZES ( ch_fasta ).sizes
 
+    //
+    // Uncompress BBSplit index or generate from scratch if required
+    //
+    ch_bbsplit_index = Channel.empty()
+    ch_bbmap_version = Channel.empty()
+    if ('bbsplit' in prepare_tool_indices) {
+        if (params.bbsplit_index) {
+            if (params.bbsplit_index.endsWith('.tar.gz')) {
+                ch_bbsplit_index = UNTAR_BBSPLIT_INDEX ( params.bbsplit_index ).untar
+            } else {
+                ch_bbsplit_index = file(params.bbsplit_index)
+            }
+        } else {
+            Channel
+                .from(file(params.bbsplit_fasta_list))
+                .splitCsv()
+                .flatMap { id, fasta -> [ [ 'id', id ], [ 'fasta', fasta ] ] }
+                .groupTuple()
+                .map { it -> it[1] }
+                .collect { [ it ] }
+                .set { ch_bbsplit_fasta_list }
```

Comment:
I find this channel transformation quite hard to follow, maybe add a small comment what it does.

Reply:
Howzzzzzat?! https://github.com/nf-core/rnaseq/pull/700/commits/1c7f9fc200ed33a665b96525c23cbc8796d91f2a

Reply:
sweet